src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs

```diff
@@ -142,150 +140,186 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
                 }
             }
+
+            attributeNames = attributes.Select(GetAttributeName).ToSet();
+            return elementName != null;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForSymbol(ISymbol symbol, DocumentationCommentTriviaSyntax trivia)
+        private string GetElementNameAndAttributes(SyntaxNode node, out SyntaxList<XmlAttributeSyntax> attributes)
         {
-            if (symbol is IMethodSymbol method)
-            {
-                return GetTagsForMethod(method, trivia);
-            }
-
-            if (symbol is IPropertySymbol property)
-            {
-                return GetTagsForProperty(property, trivia);
-            }
+            XmlNameSyntax nameSyntax;
 
-            if (symbol is INamedTypeSymbol namedType)
+            switch (node.Kind())
             {
-                return GetTagsForType(namedType, trivia);
+                case SyntaxKind.XmlEmptyElement:
+                    var emptyElementSyntax = (XmlEmptyElementSyntax)node;
+                    nameSyntax = emptyElementSyntax.Name;
+                    attributes = emptyElementSyntax.Attributes;
+                    break;
+
+                case SyntaxKind.XmlElement:
+                    node = ((XmlElementSyntax)node).StartTag;
+                    goto case SyntaxKind.XmlElementStartTag;
+
+                case SyntaxKind.XmlElementStartTag:
+                    var startTagSyntax = (XmlElementStartTagSyntax)node;
+                    nameSyntax = startTagSyntax.Name;
+                    attributes = startTagSyntax.Attributes;
+                    break;
+
+                default:
+                    nameSyntax = null;
+                    attributes = default(SyntaxList<XmlAttributeSyntax>);
+                    break;
             }
 
-            return SpecializedCollections.EmptyEnumerable<CompletionItem>();
+            return nameSyntax?.LocalName.ValueText;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForType(INamedTypeSymbol symbol, DocumentationCommentTriviaSyntax trivia)
+        private bool IsAttributeValueContext(SyntaxToken token, out string tagName, out string attributeName)
         {
-            var items = new List<CompletionItem>();
-
-            var typeParameters = symbol.TypeParameters.Select(p => p.Name).ToSet();
+            XmlAttributeSyntax attributeSyntax = null;
 
-            RemoveExistingTags(trivia, typeParameters, x => AttributeSelector(x, TypeParamTagName));
+            if (token.IsParentKind(SyntaxKind.IdentifierName) && token.Parent.IsParentKind(SyntaxKind.XmlNameAttribute))
+            {
+                // name="bar$$
+                attributeSyntax = (XmlNameAttributeSyntax)token.Parent.Parent;
+            }
+            else if (token.IsKind(SyntaxKind.XmlTextLiteralToken) && token.IsParentKind(SyntaxKind.XmlTextAttribute))
+            {
+                // foo="bar$$
```

Comment:
This is a generic non-`name` attribute. The case above handles the special `XmlNameAttributeSyntax` nodes. The node kinds and structures are different, presumably for the semantic highlighting of the names.